frontend/app/src/components/common/ListItem.vue

```diff
@@ -0,0 +1,77 @@
+<script setup lang="ts">
+const props = withDefaults(
+  defineProps<{
+    size?: 'sm' | 'md' | 'lg';
+    title?: string;
+    subtitle?: string;
+    noPadding?: boolean;
+    noHover?: boolean;
+    loading?: boolean;
+  }>(),
+  {
+    size: 'sm',
+    title: '',
+    subtitle: '',
+    noPadding: false,
+    noHover: false,
+    loading: false
+  }
+);
+
+const avatarSizeClasses = computed(() => {
+  if (props.size === 'md') {
+    return 'w-10 h-10';
+  } else if (props.size === 'lg') {
+    return 'w-12 h-12';
+  }
+  return 'w-8 h-8';
+});
+</script>
+
+<template>
+  <div
+    class="w-full flex items-center py-2 gap-2 cursor-pointer"
+    :class="{
+      'px-4': !noPadding,
+      'hover:bg-rui-grey-100 hover:dark:bg-rui-grey-800': !noHover
+    }"
+    v-on="
+      // eslint-disable-next-line vue/no-deprecated-dollar-listeners-api
+      $listeners
+    "
+  >
+    <div
+      v-if="$slots.avatar"
```

Comment:
thought we always prefer to use useSlots

Reply:
So this in mostly because of a misunderstanding I had on what gets removed.

Reply:
So I should change it to use `useSlots` right? @kelsos

Reply:
Anything is fine we can figure out what makes sense after we finish the upgrade